### PR TITLE
Fixes two out-of-bounds bugs in the InChI wrapper

### LINKED ISSUE #8129: Two out-of-bounds bugs in inchi interface
**Describe the bug**

We (https://patents.google.com/) encountered some out of bounds bugs while converting the following SMILES strings to InChI:

* `CC[C@H]1OC(=O)C[C@@H](O)[C@H](C)[C@@H](O[C@@H]2O[C@H](C)[C@@H](O)C(N(C)C)C2O)[C@@H](CC=O)C[C@@H](C)C(=O)/C=C/C(C)=C/[C@@H]1CO.CC[C@H]1OC(=O)C[C@@H](O)[C@H](C)[C@@H](O[C@@H]2O[C@H](C)[C@@H](O[C@H]3CC(C)(O)[C@@H](O)C(C)O3)C(N(C)C)C2O)[C@@H](CC=O)C[C@@H](C)C(=O)/C=C/C(C)=C/[C@@H]1CO[C@@H]1OC(C)[C@@H](O)[C@H](OC)C1OC.[3H][Y]([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])([3H])[3H]`  
  => "runtime error: index 20 out of bounds for type 'AT_NUM[20]' (aka 'short[20]')"
* `Cc1cc(C)c(N2CCN(c3c(C)cc(C)cc3C)C2[Ru@OH28]2(Cl)(Cl)=Cc3cc([S@SP3](=O)(=O)N(C)C)ccc3O->2C(C)C)c(C)c1`  
  => "runtime error: index 4 out of bounds for type 'AT_NUM[4]' (aka 'short[4]')"

### PR DESCRIPTION
**Pull request by greglandrum**
Inspired by #8123, which is where the tests came from.

Fixes #8129

### CODE REVIEW

**Comment by chbrown** [comment]
Great! Looks much more like it was written by someone who knows what they're doing ;)

**Comment by greglandrum** [comment]
> Great! Looks much more like it was written by someone who knows what they're doing ;)

😀 
It's pretty easy when someone else has already figured out exactly where the problem is and has test cases ready to go. 

Thanks, again, for the work you put in on this one

File: External/INCHI-API/inchi.cpp [hunk 1]
```diff
@@ -1840,11 +1840,14 @@ std::string MolToInchi(const ROMol &mol, ExtraInchiReturnValues &rv,
...
-    if (atom->getChiralTag() != Atom::CHI_UNSPECIFIED ||
-        atom->hasProp("molParity")) {
-      // we ignore the molParity if the number of neighbors are below 3
+    if (atom->getChiralTag() == Atom::ChiralType::CHI_TETRAHEDRAL_CCW ||
+        atom->getChiralTag() == Atom::ChiralType::CHI_TETRAHEDRAL_CW) {
       atom->calcImplicitValence();
-      if (atom->getNumImplicitHs() + atom->getDegree() < 3) {
+      if (auto tval = atom->getTotalValence(); tval < 3 || tval > 4) {
+        BOOST_LOG(rdWarningLog)
+            << "tetrahedral chirality on atom with <3 or >4 neighbors will be ignore"
```
Comment (ricrogz) [inline External/INCHI-API/inchi.cpp:1848]: ignore*d*

File: External/INCHI-API/inchi.cpp [hunk 2]
```diff
@@ -1964,6 +1966,13 @@ std::string MolToInchi(const ROMol &mol, ExtraInchiReturnValues &rv,
 
     // neighbor
     unsigned int idx = inchiAtoms[atomIndex1].num_bonds;
+    // The InChI code has a max number of neighbors allowed:
+    if (idx >= MAXVAL) {
+      BOOST_LOG(rdErrorLog)
+          << " atom " << atomIndex1 << " has too many bonds: " << idx
+          << ". The InChI library supports at most " << MAXVAL - 1 << std::endl;
```
Comment (ricrogz) [inline External/INCHI-API/inchi.cpp:1973]: Why `MAXVAL - 1` ?  the neighbors array is defined as `AT_NUM  neighbor[MAXVAL];`, so there should be space for `MAXVAL` neighbors (no null terminators here).

**Review by ricrogz — APPROVED** [review]
Just the "ignored" typo, and the question about the number of bonds. Looks good otherwise.

**Review by ricrogz — APPROVED** [review]
LGTM!

**Pull request merged**